assertj-core/src/main/java/org/assertj/core/internal/Strings.java

```diff
@@ -458,7 +464,7 @@ private static String normalizeWhitespace(CharSequence toNormalize) {
     boolean lastWasSpace = true;
     for (int i = 0; i < toNormalize.length(); i++) {
       char c = toNormalize.charAt(i);
-      if (isWhitespace(c)) {
+      if (isWhitespace(c) || isBreakingSpace(c)) {
```

Comment:
I didn't change the `isWhiteSpace` method because the `isNotBlank` assertion concider breakingSpaces as regular character and uses `isWhiteSpace` method. 

If there are more methods that need to concider that breakingSpace are the same as regular ones, we should concider doing a dedicated methods like "isRegularWhiteSpace" and "isWhiteSpace" which could includes breakingSpaces 🤔

Reply:
My comment was very confused sorry "^^

I was concidering defining a custom method named `isWhitespace`  that could be used instead of `Character::isWhitespace` but I was concerned that doing so would break the `isNotBlank` assertion (if all `Character::isWhitespace` call are replaced with a custom  `isWhitespace`  method defined as:

```java  
private static boolean isWhitespace(char c) {
    return Character.isWhitespace(c) || NON_BREAKING_SPACES.contains(c);
  }
```
)

So I concidered adding an other method called `isRegularWhiteSpace` to not break assertions like `isNotBlank` to have a method when we need to concider non-breaking spaces and an other when we need to ignore them.

```java  
private static boolean isRegularWhitespace(char c) {
    return Character.isWhitespace(c);
  }
```

Reply:
For now, let's keep the implementation as it is. If we get feedback that more should be changed, we can think about how to refactor this area.